workflow/controller/operator.go

```diff
@@ -1943,10 +1943,24 @@ func (woc *wfOperationCtx) executeTemplate(ctx context.Context, nodeName string,
 	}
 
 	node = woc.wf.GetNodeByName(node.Name)
+	if node == nil {
+		err := fmt.Errorf("no Node found by the name of %s;  wf.Status.Nodes=%+v", node.Name, woc.wf.Status.Nodes)
+		woc.log.Error(err)
+		woc.markWorkflowError(ctx, err)
+		return node, err
+	}
```

Comment:
I'm no expert on this code, but I assume it should error if node is nil. ? If not, I will move the error to where we try to reference node below.

Reply:
I think here is fine

looking at the original commit
https://github.com/argoproj/argo-workflows/pull/1552/files#diff-f321d4af83fcf8311dc80c0d50c59ac4c240f761206e7bb652709870eb9feb43R1224

The purpose of getting the node here is to retrieve node info after executing node base on template (step/dag..)

One way to look at this is Retry should be based on some kind of node template, so in theory, if there's no template there's no retry.

Reply:
I think here is fine

looking at the original commit
https://github.com/argoproj/argo-workflows/pull/1552/files#diff-f321d4af83fcf8311dc80c0d50c59ac4c240f761206e7bb652709870eb9feb43R1224

The purpose of getting the node here is to retrieve node info after executing node base on template (step/dag..)

One way to look at this is Retry should be based on some kind of node template, so in theory, if there's no template there's no retry.